src/bitstamp/generic_api.rs

```diff
@@ -99,4 +99,46 @@ impl ExchangeApi for BitstampApi {
                                     .to_string()],
            })
     }
+
+    /// Return the balances for each currency on the account
+    fn balances(&mut self) -> Result<Balances> {
+        let raw_response = self.return_balances()?;
+
+        let mut balances = Balances::new();
+
+        for row in raw_response.iter() {
+
+            let currency = match row.0.as_str() {
+                "usd_balance" => Currency::USD,
+                "btc_balance" => Currency::BTC,
+                "eur_balance" => Currency::EUR,
+                "xrp_balance" => Currency::XRP,
+//                "usd_reserved" => "usd_reserved",
+//                "btc_reserved" => "btc_reserved",
+//                "eur_reserved" => "eur_reserved",
+//                "xrp_reserved" => "xrp_reserved",
+//                "usd_available" => "usd_available",
+//                "btc_available" => "btc_available",
+//                "eur_available" => "eur_available",
+//                "xrp_available" => "xrp_available",
+//                "btcusd_fee" => "btcusd_fee",
+//                "btceur_fee" => "btceur_fee",
+//                "eurusd_fee" => "eurusd_fee",
+//                "xrpusd_fee" => "xrpusd_fee",
+//                "xrpeur_fee" => "xrpeur_fee",
+//                "xrpbtc_fee" => "xrpbtc_fee",
+//                "fee" => "fee",
+                _ => continue,
+            };
+
+            let amount = row.1
+                .as_str()
+                .unwrap()
+                .parse::<f64>()?;
```

Comment:
I tried that and it didn't work. I have to review the code and include the errors